### PR TITLE
add new rule for securing application insights telemetry

### PR DESCRIPTION
**Pull request by william-liebenberg**
## Reason for change (Issue, Email, conversation + reason, etc)

New rule for securing application insights telemetry

### CODE REVIEW

**Review by bradystroud — APPROVED** [review]
Looks fine but I'd get @brydeno or someone to check it out

File: rules/appinsights-authentication/appinsights-disable-local-auth.png
Comment (AlexRothwell) [inline rules/appinsights-authentication/appinsights-disable-local-auth.png:1]: The ID in the URL is visible here, we probably want that blurred to match the other image

File: rules/appinsights-authentication/rule.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+To preventing unintended access, Local authentication (typically Instrumentation Key or API Keys) can be **disabled** for authenticating data ingestion. Once disabled, a **stronger authentication** method known as Role **Based Access Control (RBAC)** can be used.
+
+<!--endintro-->
+
+With Role-Based Access Control (RBAC) we can define the specific applications and users that have the ability to send telemetry data, making unauthorized data injection harder.
+
+This method also allows for more accurate Monitoring and Auditing because it allows us to log who accessed our resources and what actions they performed, which is crucial for security.
+
+In summary, disabling local authentication and relying on Azure AD Role Based Access Control for telemetry ingestion in Azure Application Insights allows for more robust security measures, ensuring that only authorized entities can submit telemetry data.
```
Comment (AlexRothwell) [inline rules/appinsights-authentication/rule.md:32]:
```suggestion
In summary, disabling local authentication and relying on Entra ID RBAC for telemetry ingestion in Azure Application Insights allows for more robust security measures, ensuring that only authorized entities can submit telemetry data.
```

File: rules/appinsights-authentication/rule.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+![Increase Security - Disable Local Authentication](appinsights-disable-local-auth.png)
+
+In short, to secure the telemetry sent from your application to Application Insights, follow these steps:
+
+1. Use System-assigned or User-assigned Managed Identity with AppService
+2. Disable Local Authentication to enable RBAC
+3. Assign `Monitoring Metrics Publisher` role to the AppService
```
Comment (AlexRothwell) [inline rules/appinsights-authentication/rule.md:40]:
```suggestion
3. Assign the `Monitoring Metrics Publisher` role to the AppService
```

Comment (AlexRothwell) [inline rules/appinsights-authentication/rule.md:40]:
```suggestion
3. Assign the `Monitoring Metrics Publisher` role to the AppService
```

File: rules/appinsights-authentication/rule.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+::: good
+Figure: Assigning Monitoring Metrics Publisher role to AppService Managed Identity using Bicep
+:::
+
+![Figure: AppInsights shows the Monitoring Metrics Publisher role is assigned to AppService](appinsights-assigned-role.png)
+
+## Authenticating with Application Insights
+
+For ASP.NET Core applications running on Azure AppServices with the `Monitoring Metrics Publisher` role assigned, we can the use a managed identity (System-assigned or User-assigned) to authenticate the application with a particular AppInsights instance.
```
Comment (AlexRothwell) [inline rules/appinsights-authentication/rule.md:146]:
```suggestion
For ASP.NET Core applications running on Azure AppServices with the `Monitoring Metrics Publisher` role assigned, we can use a managed identity (System-assigned or User-assigned) to authenticate the application with a particular AppInsights instance.
```

File: rules/appinsights-authentication/rule.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+---
+
+Azure Application Insights is your app's friendly doctor, keeping an eye on it to stay healthy and running smoothly. It uses an `Instrumentation Key` to allow your app to send telemetry data to its resource.
+
+While this `Instrumentation Key` isn't a secret and is often included in client-side code, it's crucial to keep it safe. If it falls into the wrong hands, it could corrupt your data or lead to extra costs.
+
+The key doesn't grant access to your data; for that, you use Entra ID (used to be Azure AD) authentication.
+
+To preventing unintended access, Local authentication (typically Instrumentation Key or API Keys) can be **disabled** for authenticating data ingestion. Once disabled, a **stronger authentication** method known as Role **Based Access Control (RBAC)** can be used.
```
Comment (christoment) [inline rules/appinsights-authentication/rule.md:24]: This is a great rule, but I can see this rule can be used in the unintended direction.

I think we should clearly outline the trade-off of securing application insights.
Enabling this effectively blocks sending any client-side telemetries to this AI instance - this approach will require us to do an alternative such as:
- create another application insight and then combine them together somehow
- only use SSR frontend
- create a secure ingestion point in backend (I think this is a bad approach)
- secure frontend with user AD Authentication (locks us down to AD)

What do you think? 😄

Comment (william-liebenberg) [reply rules/appinsights-authentication/rule.md:24]: I agree with everything you listed above. There will be trade offs.

I can add the PROs and CONs for securing server-side telemetry, but I'll dig a bit further to see what we can recommend for client-side telemetry.

Thanks @christoment !

**Pull request merged**